src/main/java/org/opentripplanner/model/plan/ScheduledTransitLeg.java

```diff
@@ -215,6 +224,14 @@ protected void setDistanceMeters(double distanceMeters) {
     this.distanceMeters = DoubleUtils.roundTo2Decimals(distanceMeters);
   }
 
+  public double getDirectDistanceMeters() {
+    return directDistanceMeters;
+  }
+
+  protected void setDirectDistanceMeters(double directDistanceMeters) {
```

Comment:
Can you not compute that on the fly?

Reply:
I think it is the easiest to do it here as there is a private function `extractTransitLegCoordinateshere` that figures out the coordinates from the leg. I think computing the distance in the fare calculator would require copying this function there or making it public, which might not be wanted

Reply:
I see that there is already a method just like that. Let's discuss this in the meeting tomorrow.